### ISSUE TITLE
OP_REQUIRES failed at example_parsing_ops.cc:92 : Invalid argument: Could not parse example input

### ISSUE THREAD

**Issue by C-Compton** [tensorflow/serving#1890]
## Bug Report
If this is a bug report, please fill out the following form in full:

### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux version 5.10.25-linuxkit docker container on MacOS 14.1.1 
- **TensorFlow Serving installed from (source or binary)**: docker hub 
- **TensorFlow Serving version**: latest

### Describe the problem
Model is constructed in a TFX pipeline based on the taxi template. The features are preprocessed into two sets:
1. Vocab feature keys preprocessed with `tft.compute_and_apply_vocabulary`
2. Bucketized feature keys of floats.

Tensorflow/Serving container is started and successfully finds the pushed model. 

When making predictions, a very small subset of data will result in a successful prediction. However, if I change one of the vocabulary features to some other value that is present in the vocabulary files in the model assets directory, the model failed to parse the input. 

### Exact Steps to Reproduce
`curl -d '{"instances": ["\"SOME_VOCAB\", \"SOME_OTHER_VOCAB\", 1.3, 3.4, ..., \"ANOTHER_VOCAB\""]}' -X POST http://localhost:9501/v1/models/pushed_model:predict 
`
would succeed, but then 

`curl -d '{"instances": ["\"DIFF_VOCAB\", \"SOME_OTHER_VOCAB\", 1.3, 3.4, ..., \"ANOTHER_VOCAB\""]}' -X POST http://localhost:9501/v1/models/pushed_model:predict`
would result in 

> 021-07-07 16:48:10.226720: W external/org_tensorflow/tensorflow/core/framework/op_kernel.cc:1767] OP_REQUIRES failed at example_parsing_ops.cc:92 : Invalid argument: Could not parse example input, value: '"DIFF_VOCAB", "SOME_OTHER_VOCAB", 1.3, 3.4, ..., "ANOTHER_VOCAB"'

**Comment by arghyaganguly** [comment]
@C-Compton , please checkout similar issue in [link.](https://github.com/tensorflow/ranking/issues/189)

**Comment by arghyaganguly** [comment]
@C-Compton , please follow [link](https://colab.research.google.com/gist/hanneshapke/e6b031484ed2474165c4eccb37f69ce3/request-prediction-from-bert-model-with-tft-preprocessing.ipynb#scrollTo=2CZ9P13ioDnm) and try to encode with base64 as shown in the predict function.

**Comment by C-Compton** [comment]
@arghyaganguly  Thank you for the tip. I was able to get responses from inputs that previously resulted in the parsing error. 

I have additional questions, however. 

Your solution linked above does some additional pre-processing to the request data before posting to tensorflow/serving. Our application that is making prediction requests is not written in Python, and so cannot pre-process the requests as they are in your solution, specifically the tensorflow function calls.  Is there an alternative solution that wouldn't require these additional steps?

**Comment by C-Compton** [comment]
Just to follow up on yesterday,

I'm hoping there is a solution I'm unaware of that doesn't require tensorflow methods to preprocess request data. Converting to base64 is simple, but I feel any application ought to be able to make prediction requests without any other special treatment. Particularly since I didn't notice any indication that these steps were needed when reviewing the tensorflow/serving REST documentation. 

Are there steps one can take in the TFX pipeline, for instance, that would remove the need of them here?

**Comment by gharibian** [comment]
This is not a servo issue; your input example has a parsing error and fails at: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/core/kernels/example_parsing_ops.cc#L92

Please thoroughly inspect your altered input for invalid syntax or unexpected unicode characters.  example_parsing_ops.cc is heavily used and sees very little change; the parsing itself is done by the proto parser (MessageLite::ParseFromArray) which is used by thousands of projects.  If you have a strong reason to believe this is a bug, please provide a fully reproducible self-contained non-redacted example.

**Comment by C-Compton** [comment]
@gharibian I wasn't sure what the best way to do this was so I created a colab of outlining which files in the TFX pipeline I modified as well as two sample calls to the tensorflow/serving container; 1 succeeding, the other failing. 

https://colab.research.google.com/drive/15JDtVQYH_pd8gU-5gK0j8gO2LUTRgnmn?usp=sharing

I've also attached the training data file. 

[data.csv](https://github.com/tensorflow/serving/files/6841836/data.csv)

**Comment by C-Compton** [comment]
Hello. Any updates on this? Are the files outlined in the colab sufficient to be able to create a reproducible environment? Are the transforms, requests, etc in good form? 

Thanks.

**Comment by C-Compton** [comment]
@arghyaganguly 

I attempted to follow this blog post to create a gRPC client. 
https://medium.com/@junwan01/a-java-client-for-tensorflow-serving-grpc-api-d37b5ad747aa

I've compiled the following and run it against my model. 
`public class MainClass {

    public static void main(String[] args) {
        String host = "localhost";
        int port = 8500;
        String modelName = "pushed_model";
        int modelVersion = 36;

        String modelInput = "{\"instances\": ['\"ARCHIVED\",\"EXTERNAL\",\"DYNAMICALLY_LINKED\",\"MANUAL_BOM_FILE\",\"LOW\",4.3,2.9,3.2,8.6,\"NETWORK\",\"NONE\",\"NONE\",\"PARTIAL\",\"NONE\",\"UNPROVEN\",\"OFFICIAL_FIX\",\"CONFIRMED\",\"MEDIUM\",5.5,3.6,4.8,1.8,\"LOCAL\",\"LOW\",\"NONE\",\"HIGH\",\"NONE\",\"NONE\",\"UNCHANGED\",\"REQUIRED\",\"UNPROVEN\",\"OFFICIAL_FIX\",\"CONFIRMED\",\"{CWE-22}\"']}"
        ;

        byte[] encoded = Base64.getEncoder().encode(modelInput.getBytes());
        byte[] decoded = null;
        try {
            decoded = Base64.getDecoder().decode(new String(encoded).getBytes("UTF-8"));
        } catch(Exception e) {
            System.out.println(e);
        }
        ManagedChannel channel = ManagedChannelBuilder.forAddress(host, port).usePlaintext().build();
        PredictionServiceGrpc.PredictionServiceBlockingStub stub = PredictionServiceGrpc.newBlockingStub(channel);

        Model.ModelSpec.Builder modelSpecBuilder = Model.ModelSpec.newBuilder();
        modelSpecBuilder.setName(modelName);
        modelSpecBuilder.setVersion(Int64Value.of(modelVersion));
        modelSpecBuilder.setSignatureName("serving_default");

        Predict.PredictRequest.Builder builder = Predict.PredictRequest.newBuilder();
        builder.setModelSpec(modelSpecBuilder);

        TensorProto.Builder tensorProtoBuilder = TensorProto.newBuilder();
        tensorProtoBuilder.setDtype(DataType.DT_STRING);
        TensorShapeProto.Builder tensorShapeBuilder = TensorShapeProto.newBuilder();
        tensorShapeBuilder.addDim(TensorShapeProto.Dim.newBuilder().setSize(1));
        tensorProtoBuilder.addStringVal(ByteString.copyFrom(decoded));
        TensorProto tp = tensorProtoBuilder.build();

        builder.putInputs("inputs", tp);

        Predict.PredictRequest request = builder.build();
        Predict.PredictResponse response = stub.predict(request);

        System.out.println(response);
    }
}`

I still get parsing errors form the serving container. 

> Exception in thread "main" io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Could not parse example input, value: '{"instances": ['"ARCHIVED","EXTERNAL","DYNAMICALLY_LINKED","MANUAL_BOM_FILE","LOW",4.3,2.9,3.2,8.6,"NETWORK","NONE","NONE","PARTIAL","NONE","UNPROVEN","OFFICIAL_FIX","CONFIRMED","MEDIUM",5.5,3.6,4.8,1.8,"LOCAL","LOW","NONE","HIGH","NONE","NONE","UNCHANGED","REQUIRED","UNPROVEN","OFFICIAL_FIX","CONFIRMED","{CWE-22}"']}'
	 [[{{node ParseExample/ParseExampleV2}}]]
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:262)
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:243)
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:156)
	at tensorflow.serving.PredictionServiceGrpc$PredictionServiceBlockingStub.predict(PredictionServiceGrpc.java:441)
	at main.MainClass.main(MainClass.java:55)

Process finished with exit code 1

Are there any changes I need to make in the Java code or modelInput?

**Comment by C-Compton** [comment]
Closing this issue due to lack of support and responses. We're cutting our losses and opting for an alternative ML library.
